frontend/src/components/tablestates.js

```diff
@@ -15,58 +15,57 @@ import {
-          <EmptyStateFooter>
-            {!!this.props.onClearFilters &&
-            <React.Fragment>
-              <EmptyStateBody>
-                No results match this filter criteria. Clear all filters to show results.
-              </EmptyStateBody>
-              <EmptyStateActions>
-                <Button variant="link" onClick={this.props.onClearFilters}>Clear all filters</Button>
-              </EmptyStateActions>
-            </React.Fragment>
-            }
-          </EmptyStateFooter>
-        </EmptyState>
-      </Bullseye>
-    );
-  }
+const TableEmptyState = (props) => {
+  const {onClearFilters} = props;
+  <Bullseye>
+    <EmptyState>
+      <EmptyStateHeader titleText="No results found" icon={<EmptyStateIcon icon={SearchIcon} />} headingLevel="h5" />
+      <EmptyStateFooter>
+        {!!this.props.onClearFilters &&
+        <React.Fragment>
+          <EmptyStateBody>
+            No results match this filter criteria. Clear all filters to show results.
+          </EmptyStateBody>
+          <EmptyStateActions>
+            <Button variant="link" onClick={onClearFilters}>Clear all filters</Button>
+          </EmptyStateActions>
+        </React.Fragment>
+        }
+      </EmptyStateFooter>
+    </EmptyState>
+  </Bullseye>
 }
 
-export class TableErrorState extends React.Component {
-  static propTypes = {
-    onClearFilters: PropTypes.func,
-  }
+TableEmptyState.propTypes = {
+  onClearFilters: PropTypes.func,
+}
+
+const TableErrorState = (props) => {
+  const {onClearFilters} = props;
 
-  render() {
-    return (
-      <Bullseye>
-        <EmptyState>
-          <EmptyStateHeader titleText="Error occurred fetching results" icon={<EmptyStateIcon icon={ErrorCircleOIcon} />} headingLevel="h5" />
-          <EmptyStateFooter>
-            {!!this.props.onClearFilters &&
-            <React.Fragment>
-              <EmptyStateBody>
-                An error occurred while fetching results. Try a different set of filters.
-              </EmptyStateBody>
-              <EmptyStateActions>
-                <Button variant="link" onClick={this.props.onClearFilters}>Clear all filters</Button>
-              </EmptyStateActions>
-            </React.Fragment>
-            }
-          </EmptyStateFooter>
-        </EmptyState>
-      </Bullseye>
-    );
-  }
+  <Bullseye>
+    <EmptyState>
+      <EmptyStateHeader titleText="Error occurred fetching results" icon={<EmptyStateIcon icon={ErrorCircleOIcon} />} headingLevel="h5" />
+      <EmptyStateFooter>
+        {!!this.props.onClearFilters &&
+        <React.Fragment>
+          <EmptyStateBody>
+            An error occurred while fetching results. Try a different set of filters.
+          </EmptyStateBody>
+          <EmptyStateActions>
+            <Button variant="link" onClick={onClearFilters}>Clear all filters</Button>
+          </EmptyStateActions>
+        </React.Fragment>
+        }
+      </EmptyStateFooter>
+    </EmptyState>
+  </Bullseye>
```

Comment:
Both `TableErrorState` & `TableEmptyState` are almost identical, I think we can merge these 2 and add more props to differ them:

```
const TableState = ({ title, icon, message, onClearFilters }) => (
...
);

const TableEmptyState = (props) => (
  <TableState
    title="No results found"
    icon={SearchIcon}
    message="No results match this filter criteria. Clear all filters to show results."
    {...props}
  />
);

const TableErrorState = (props) => (
  <TableState
    title="Error occurred fetching results"
    icon={ErrorCircleOIcon}
    message="An error occurred while fetching results. Try a different set of filters."
    {...props}
  />
);
```

Reply:
yeah, I put a TODO above the first component mentioning this, but don't want to do it right now. :)